expr_checker.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import ast
+import types
+from inspect import cleandoc, getsource
+
+
+def __ast_default_check_type(method, value):
+    safe_type = (
+        str, bytes, float, complex, int, bool, types.NoneType,
+        tuple, list, set, dict,
+        range,
+        types.FunctionType, types.LambdaType, types.GeneratorType, types.MethodType,
+        types.BuiltinFunctionType, types.BuiltinMethodType, types.WrapperDescriptorType,
+        types.MethodWrapperType, types.MethodDescriptorType, types.ClassMethodDescriptorType,
+        types.ModuleType
```

Comment:
for those types we can be a little more specific I think
- they can not be returned by function
- they can be returned by "getattribute"
- they can not be use in `check_type('called', func.__self__)`
- they can not be given in function argument

We only need them to be able to do "datetime.datetime.now()" e.g.

can you make a separate list and check them only where it's needed ? (+ write some tests about that)